### DISCUSSION TITLE
TkCollectionView invalid batch update

### DISCUSSION THREAD

**Discussion by ad-proximate** [NativeScript/NativeScript#10344] [Q&A]
I'm using RadListView in my project. Although I haven't changed the list view in the last weeks the application started to throw the following error:

`ERROR Error: Invalid batch updates detected: the number of sections and/or items returned by the data source before and/or after performing the batch updates are inconsistent with the updates.

Data source before updates = { 1 section with item counts: [4] }

Data source after updates = { 1 section with item counts: [4] }

Updates = [

Insert item (0 - 0),

Insert item (0 - 1),

Insert item (0 - 2)

]

Collection view: <TKCollectionView: 0x10498ae00; baseClass = UICollectionView; frame = (0 0; 375 406.333); clipsToBounds = YES; autoresize = W+H; gestureRecognizers = <NSArray: 0x2814d2df0>; backgroundColor = UIExtendedGrayColorSpace 0 0; layer = <CALayer: 0x281af2380>; contentOffset: {0, 0}; contentSize: {936, 406.33333333333331}; adjustedContentInset: {0, 0, 0, 0}; layout: <TKListViewLinearLayout: 0x11dccbcd0>; dataSource: <TKListView: 0x11dccdaf0; frame = (0 45.3333; 375 406.333); autoresize = W+H; tag = 3128; backgroundColor = UIExtendedSRGBColorSpace 0 0 0 0; layer = <CALayer: 0x281af192<\M-b\M^@\M-&>`

 Anybody knows how to catch/solve this problem. What might be the root of it?

**Comment by switchstance-devs** [comment]
@ad-proximate Have you managed to resolve this issue? For me it occurs when running the app on iOS > 16.4 but no issue when running on iOS 16.2. Upgrading to most recent version of nativescript/ios didn't help.
